src/mam4xx/hetfrz.hpp

```diff
@@ -0,0 +1,1490 @@
+
+  const Real coeff_vlc_a = 8.8462e2;
+  const Real coeff_vlc_b = 9.7593e7;
+  const Real coeff_vlc_c = -3.4249e-11;
+  const Real coeff_adj_a = 3.1250e-1;
+  const Real coeff_adj_b = 1.0552e-3;
+  const Real coeff_adj_c = -2.4023;
+
+  // droplet terminal velocity after Chen & Liu, QJRMS 2004
+  const Real vlc_drop_adjfunc = haero::exp(
+      haero::exp(coeff_adj_a + coeff_adj_b * haero::cube(haero::log(r3lx)) +
+                 coeff_adj_c * haero::pow(rho_air, 1.5)));
+  const Real vlc_drop =
+      (coeff_vlc_a + (coeff_vlc_b + coeff_vlc_c * r3lx) * r3lx) * r3lx *
+      vlc_drop_adjfunc;
+
+  // Reynolds number
+  const Real Re = 2 * vlc_drop * r3lx * rho_air / viscos_air;
+
+  return Re;
+}
+
+KOKKOS_INLINE_FUNCTION
+Real get_temperature_diff(const Real temperature, const Real pressure,
+                          const Real eswtr, const Real latvap,
+                          const Real Ktherm_air) {
+
+  // water vapor diffusivity: Pruppacher & Klett 13-3
+  const Real Dvap = 0.211e-4 * (temperature / 273.15) * (101325. / pressure);
+
+  const Real rhoh2o = Constants::density_h2o;
+  const Real rh2o =
+      461.50463982015992; // (BAD CONSTANT) water vapor gas constant
+
+  // G-factor = rhoh2o*Xi in Rogers & Yau, p. 104
+
+  const Real G_factor = rhoh2o / ((latvap / (rh2o * temperature) - 1) * latvap *
+                                      rhoh2o / (Ktherm_air * temperature) +
+                                  rhoh2o * rh2o * temperature / (Dvap * eswtr));
+
+  // calculate T-Tc as in Cotton et al.
+  const Real Tdiff =
+      -G_factor * (Hetfrz::rhwincloud - 1.) * latvap / Ktherm_air;
+
+  return Tdiff;
+}
+
+KOKKOS_INLINE_FUNCTION
+void calculate_collkernel_sub(const Real temperature, const Real pressure,
+                              const Real rho_air, const Real r3lx,
+                              const Real r_a, const Real lambda,
+                              const Real latvap, const Real viscos_air,
+                              const Real Re, const Real Ktherm_air,
+                              const Real Ktherm, const Real Pr, Real Tdiff,
+                              Real &K_total) {
+
+  // Note for C++ port: Due to BFB for Fortran code, we have to declare
+  // Boltzmann constant here again. This value is only used in this
+  // subroutine.
+  constexpr Real kboltz2 = 1.38065e-23;                // BAD CONSTANT
+  constexpr Real pi = 3.1415926535897931;              // BAD CONSTANT
+  constexpr Real r_gas_h2o_vapor = 461.50463982015992; // BAD CONSTANT
+  // Knudsen number (Seinfeld & Pandis 8.1)
+  const Real Kn = lambda / r_a;
+
+  // aerosol diffusivity
+  const Real Daer =
+      kboltz2 * temperature * (1 + Kn) / (6.0 * pi * r_a * viscos_air);
+
+  // Schmidt number
+  const Real Sc = viscos_air / (Daer * rho_air);
+  //  Young (1974) first equ. on page 771
+  // Todo (the 0.33 power here should probably be 1/3, but that wouldn't be
+  // BFB.)
+
+  const Real K_brownian = 4.0 * pi * r3lx * Daer *
```

Comment:
I would change 0.33 to 1./3.

Reply:
BFB is nice to have at this particular moment, so maybe we can add this to a GitHub issue to revisit later. I don't remember whether we have a "things to go back and fix" issue (I think not), but we could start anytime!

Reply:
This is a direct port from F90, so switching from 0.33 to 1.0/.30 or using `haero::cbrt()` would change the solution.